src/huggingface_hub/hf_api.py

```diff
@@ -3542,6 +3541,39 @@ def update_repo_visibility(
         hf_raise_for_status(r)
         return r.json()
 
+    @validate_hf_hub_args
+    def update_repo_settings(
```

Comment:
I think this method should be renamed to something like `update_repo_gated_status`. imo, It better describes the function's purpose of modifying a repo's gated access status.

Reply:
Yeah, `update_repo_gated_status` this will have more clarity

Reply:
Hi, sorry the naming comes from me in https://github.com/huggingface/huggingface_hub/issues/2447#issue-2465284588 and I haven't provided the context for it. Repos have multiple settings (gating, visibility, discussion tab, etc.) and I think it'll be nice to have a single helper to handle them all. Defining the method `update_repo_settings` in the PR to handle gating is the first step towards this. I think it'll make sense in a future PR to deprecate `update_repo_visibibility` and/or make it an alias of this new one.

Reply:
Makes sense! 👍